operator/internal/controller/usersignup/usersignup_controller.go

```diff
@@ -82,8 +82,11 @@ func (r *UserSignupReconciler) ensureWorkspaceIsPresentForHomeSpace(ctx context.
 		},
 	}
 	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, w, func() error {
-		w.Spec.DisplayName = "default"
-		w.Spec.Visibility = workspacesv1alpha1.InternalWorkspaceVisibilityPrivate
+		if w.ObjectMeta.CreationTimestamp.IsZero() {
```

Comment:
What is this check doing, making sure this is not an update event?

Reply:
exactly, this way we set this fields only when creating the resource. Otherwise we will reset visibility and display name of the workspace every time an UserSignup is updated/reconciled

Reply:
We could hoist these assignments out of this closure.  They'll be overwritten if the resource exists and has those fields set.

Reply:
ack, then this is fine by me.